packages/esm-generic-patient-widgets-app/src/obs-table-horizontal/obs-table-horizontal.component.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { usePagination, useConfig, formatDate, formatTime } from '@openmrs/esm-framework';
+import {
+  Table,
+  TableCell,
+  TableContainer,
+  TableBody,
+  TableHead,
+  TableHeader,
+  TableRow,
+  InlineLoading,
+} from '@carbon/react';
+import { CardHeader, PatientChartPagination } from '@openmrs/esm-patient-common-lib';
+import { useObs } from '../resources/useObs';
+import styles from './obs-table-horizontal.scss';
+import { useTranslation } from 'react-i18next';
+import { type ConfigObjectHorizontal } from '../config-schema-obs-horizontal';
+
+interface ObsTableHorizontalProps {
+  patientUuid: string;
+}
+
+const ObsTableHorizontal: React.FC<ObsTableHorizontalProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const config = useConfig<ConfigObjectHorizontal>();
+  const { data: obss, isValidating } = useObs(patientUuid, config.showEncounterType);
+  const uniqueEncounterReferences = [...new Set(obss.map((o) => o.encounter.reference))].sort();
+  let obssGroupedByEncounters = uniqueEncounterReferences.map((reference) =>
+    obss.filter((o) => o.encounter.reference === reference),
+  );
+
+  if (config.oldestFirst) {
+    obssGroupedByEncounters.sort(
+      (a, b) => new Date(a[0].effectiveDateTime).getTime() - new Date(b[0].effectiveDateTime).getTime(),
+    );
+  } else {
+    obssGroupedByEncounters.sort(
+      (a, b) => new Date(b[0].effectiveDateTime).getTime() - new Date(a[0].effectiveDateTime).getTime(),
+    );
+  }
+
+  let tableRowLabels = config.data.map(({ concept, label }) => ({
+    key: concept,
+    header: label || obss.find((o) => o.conceptUuid === concept)?.code?.text,
```

Comment:
I think it would be good if we ran the `label` through the `t` function. While someone obviously can't provide a translation in the code, translations could be provided via the "Translation overrides",  which would allow this to better support multi-lingual instances.